src/map_io/map_players_view_packet.cc

```diff
@@ -385,8 +385,8 @@ void MapPlayersViewPacket::read(FileSystem& fs, EditorGameBase& egbase) {
 					f.resource_amounts.d = fr.unsigned_8();
 					f.resource_amounts.r = fr.unsigned_8();
 
-					f.terrains.d = fr.unsigned_32();
-					f.terrains.r = fr.unsigned_32();
+					f.terrains.store({static_cast<DescriptionIndex>(fr.unsigned_32()),
+					                  static_cast<DescriptionIndex>(fr.unsigned_32())});
```

Comment:
Calling several function of FileRead in the same statement is dangerous, please store them in local variables first. 
Guaranteed execution order? Some compilers are above that ;)